data/transactions/logic/opcodes.go

```diff
@@ -65,25 +65,34 @@ const fidoVersion = 7 // base64, json, secp256r1
 // arguments, or dynamic layout controlled by a check function.
 type opDetails struct {
 	Cost       int
+	costFunc   opCostFunc
```

Comment:
@jannotti For my understanding:  What tradeoffs were considered when introducing `costFunc` as a new parameter?

* In my imagination, it's possible to _replace_ `Cost` with `costFunc`.  Doing so, appears to provide these benefits:
  * A single mental model for considering an opcode's costs.  The PR's approach requires mental and call-site branching.
  * Makes it possible to return an explicit error when a cost can't be resolved.  (Mostly) addresses the concern around returning a 0 cost.  Though we may still wish to make a 0 check.
* I'm imagining rationale for _not_ making the change may involve a backwards compatibility concern or a (perceived) performance issue (function call vs int).

Reply:
I didn't want to slow down the hot loop of AVM evaluation.  For the vast majority of cases, getting a simple integer suffices, and I only need to make the (relatively expensive) indirect function call for opcodes that are expensive anyway.

Reply:
Summarizing a verbal with @jannotti here:
* One option that _may_ be explored is:  Find a way to represent opcode definition in such a way that an inline _cost calculator_ function can be created.  The imagined functions replaces `Cost` and `costFunc` using the spec's details to _efficiently_ compute cost.
* If the above option isn't in the cards for the PR, I optionally requested an inline comment explaining the performance concern for future maintainers.

I'll leave the thread open until a conclusion is reached.

Reply:
I'll throw in a comment for now, I think the use a computeCost() function that works off data in the OpSpec is probably the way to go if we want to maintain the speed we have now. (as opposed to what I did here, which is an indirect function call whenever cost=0)

Reply:
@jannotti Thanks for the comment in https://github.com/algorand/go-algorand/pull/3832/commits/8256400034635c50456ef3891fa25f244d3d007b - It captures my feedback.